packages/react/src/components/radio-group/radio-group.tsx

```diff
@@ -1,35 +1,87 @@
+  createFocusPropsSelector,
+  createRadioGroupStore,
+  createRootContainerPropsSelector,
+  getFocusValue,
+  getRootContainerDomOptionsFromProps,
+  RADIO_GROUP_ACTIONS,
+} from '@devextreme/components';
+import {
+  Children,
+  cloneElement,
+  isValidElement,
+  memo,
+  PropsWithChildren,
+  useCallback,
+  useMemo,
 } from 'react';
-import { useCallbackRef, useSecondEffect } from '../../internal/hooks';
-import { EditorProps } from '../../internal/props';
+import { FocusableRootContainer } from '../../internal/components';
+import { useCallbackRef, useSecondEffect, useStoreSelector } from '../../internal/hooks';
+import { EditorProps, RootContainerProps } from '../../internal/props';
 import { RadioGroupStoreContext } from '../radio-common';
 
+import '@devextreme/styles/src/radio-group/radio-group.scss';
+
 function RadioGroupInternal<T>(props: RadioGroupProps<T>) {
-  const controlledMode = useMemo(() => Object.hasOwnProperty.call(props, 'value'), []);
+  const isValueControlled = useMemo(() => Object.hasOwnProperty.call(props, 'value'), []);
+  const isFocusControlled = useMemo(() => Object.hasOwnProperty.call(props, 'focus'), []);
   const valueChange = useCallbackRef(props.valueChange);
+  const focusChange = useCallbackRef(props.focusChange);
 
   const store = useMemo(() => createRadioGroupStore<T>({
-    value: controlledMode ? props.value : props.defaultValue,
+    readonly: getRootContainerDomOptionsFromProps(props),
+    value: isValueControlled ? props.value : props.defaultValue,
+    focus: isFocusControlled
+      ? getFocusValue(props.focus, props.disabled)
+      : getFocusValue(props.defaultFocus, props.disabled),
   }, {
     value: {
-      controlledMode,
+      controlledMode: isValueControlled,
       changeCallback: (value: T) => { valueChange.current(value); },
     },
+    focus: {
+      controlledMode: isFocusControlled,
+      changeCallback: (focus: boolean) => { focusChange.current(focus); },
+    },
   }), []);
 
+  const rootContainerProps = useStoreSelector(store, createRootContainerPropsSelector());
+  const focusProps = useStoreSelector(store, createFocusPropsSelector());
+
+  const focusHandler = useCallback(() => {
+    store.addUpdate(RADIO_GROUP_ACTIONS.updateFocus(true));
+    store.commitUpdates();
+  }, []);
+  const blurHandler = useCallback(() => {
+    store.addUpdate(RADIO_GROUP_ACTIONS.updateFocus(false));
+    store.commitUpdates();
+  }, []);
+
   useSecondEffect(() => {
-    if (controlledMode) {
+    if (isValueControlled) {
       store.addUpdate(RADIO_GROUP_ACTIONS.updateValue(props.value));
     }
+    if (isFocusControlled) {
+      store.addUpdate(RADIO_GROUP_ACTIONS.updateFocus(getFocusValue(props.focus, props.disabled)));
+    }
+
+    const updateUiValue = getRootContainerDomOptionsFromProps(props);
+    store.addUpdate(RADIO_GROUP_ACTIONS.updateUi(updateUiValue));
 
     store.commitPropsUpdates();
-  }, [props.value]);
+  }, [props]);
```

Comment:
The effect will be called on every prop change, so, even if the value has not changed, the updateValue will be called. It is better to split it into several separate useEffect hooks, one for value, one for focus and one for UI props.

Reply:
I tried to fix it in the latest version of the PR.
Please take a look :)